aks/application/data.tf

```diff
@@ -0,0 +1,19 @@
+data "azurerm_resource_group" "monitoring" {
+  count = var.azure_enable_container_monitoring ? 1 : 0
+
+  name = "${var.azure_resource_prefix}-${var.service_short}-mn-rg"
+}
+
+data "azurerm_monitor_action_group" "main" {
+  count = var.azure_enable_container_monitoring ? 1 : 0
+
+  name                = "${var.azure_resource_prefix}-${var.service_name}"
+  resource_group_name = data.azurerm_resource_group.monitoring[0].name
+}
```

Comment:
These same lines will appear in 3 places in this repo after this change, so it makes me wonder if it's time to reduce the duplication here and provide a "monitoring" module which encapsulates this logic in once place and then exposes the information needed as outputs (similar to the cluster data module).

https://github.com/DFE-Digital/terraform-modules/blob/83801213853ed1e4b4bdcb8d36773c8683ff010f/aks/postgres/data.tf#L22-L33

https://github.com/DFE-Digital/terraform-modules/blob/83801213853ed1e4b4bdcb8d36773c8683ff010f/aks/redis/data.tf#L22-L33

Reply:
Have added this to the platform backlog 'Terraform modules improvements'.